ecommerce/views_test.py

```diff
@@ -1221,19 +1221,58 @@ def test_products_viewset_list(user_drf_client, coupon_product_ids):
         )
 
 
+def test_products_viewset_list_nested_sort(user_drf_client):
+    """ Test that the ProductViewSet returns all products sorted by (parent.title, title) when requested"""
+    product_version_1 = ProductVersionFactory.create(
+        product__content_object__title="C", product__content_object__course__title="B"
+    )
+    product_version_2 = ProductVersionFactory.create(
+        product__content_object__title="E", product__content_object__course__title="A"
+    )
+    product_version_3 = ProductVersionFactory.create(
+        product__content_object__title="D", product__content_object__course__title="C"
+    )
+    product_version_4 = ProductVersionFactory.create(
+        product__content_object__title="B", product__content_object__course__title="D"
+    )
+
+    # In alphabetical orders per product__content_object__title
+    product_versions_sorted = [
+        product_version_2,
+        product_version_1,
+        product_version_3,
+        product_version_4,
+    ]
+    product_ids_sorted = [
+        product_version.product.id for product_version in product_versions_sorted
+    ]
+
+    response = user_drf_client.get(reverse("products_api-list"))
+    assert response.status_code == status.HTTP_200_OK
+    products = response.json()
+    assert [product.get("id") for product in products] != product_ids_sorted
+
+    response_sorted = user_drf_client.get(
+        reverse("products_api-list") + "?sort=title&nested=false"
+    )
+    assert response_sorted.status_code == status.HTTP_200_OK
+    products = response_sorted.json()
+    assert [product.get("id") for product in products] == product_ids_sorted
```

Comment:
This is pretty verbose now that it's two test cases. We can clean this up and turn it into a single test case:

```python
@pytest.mark.parametrize("nested", [False, True])
def test_products_viewset_list_sort(user_drf_client, nested):
    product_versions = ProductVersionFactory.create_batch(
        4,
        product__content_object__title=factory.Iterator(["C", "E", "D", "B"]), 
        product__content_object__course__title=factory.Iterator(["B", "A", "C", "D"]),
    )
    product_versions_sorted_flat = [
       # ...
    ]
    product_versions_sorted_nested = [
       # ...
    ]
    response_sorted = user_drf_client.get(
         "{}?sort=title&nested={}".format(reverse("products_api-list"), nested)
    )
    assert response_sorted.status_code == status.HTTP_200_OK
    resp_products = response_sorted.json()
    expected_products = product_versions_sorted_nested if nested else product_versions_sorted_flat
    assert [product.get("id") for product in resp_products] == [product.id for product in expected_products]

```

We don't need the assertion that the response without `sort=title` is unsorted

Reply:
Updated